docs/messages/README.md

```diff
@@ -17,33 +17,99 @@ The protocol comprises various messages, which can be classified as either _requ
   - **WITHDRAW_FLAVOR**, sent by the provider to the consumer to notify that a specific flavor is no longer available. This message serves as a notification mechanism to inform the consumer that the requested flavor is no longer available.
 
+# AUTHENTICATION
+
+The REAR protocol foresees a peer-to-peer interaction between the **consumer** (i.e., the node requesting a given resource) and the **provider** (i.e., the node offering the reaource). 
+Both parties must be authenticated to prevent unauthorized access to the resources available in the continuum.
+
+At it's core, REAR relies on a **Decentralized identifiers (DIDs)** system as a way to provide globally unique identifier to verify the various actors involved in the REAR message exchange and removing the need for a centralized registry.
+Each device in the continuum is represented by a set of attributes, constituting what's called a **Verifiable Credential (VC)**.
+Upon connecting to the continuum, a node must performe the enrolment towards a Trusted Issuer to be included in the DID, receiving as the VC with the attributes disclodesed during the enrolment (see Figure 2).  
+
+![](/images/REAR_auth.png)
+
+_Figure 2. Example of interaction between consumer and trusted issuer_
+
+In the following, we outline the fields that can be included in the VC (note that no all attributes must be present):
+
+ - "holderName": "John Doe"
+ - "name": "John Doe"
+ - "phone": "+1-555-1234"  
+ - "holderRole": "Researcher"
+ - "fluidosRole": "Admin"
+ - "association": "IEEE"
+ - "deviceType": "Laptop"
+ - "fluidosID": "12345ABC"
+ - "holderAddress": "1234 Elm Street, Apt 5"
+ - "holderEmail": "john.doe@example.com"  
+ - "ou": "Research and Development"
+ - "physicalAddress": "5678 Maple Avenue, Building B"
+ - "DID": "did:example:123456789abcdefghi"
+ - "macAddress": "00:1A:2B:3C:4D:5E"
+ - "registrationDate": "2023-01-15T10:00:00Z"
+ - "orgIdentifier": "ORG-123456"
+ - "department": "Computer Science"
+ - "university": "Tech University"
+ - "organization": "Tech Innovations Inc."
+ - "city": "San Francisco"
+ - "zipcode": "94107" 
+ - "state": "California"
+ - "province": "California"
+ - "country": "USA"
+ - "dn": "CN=John Doe, OU=Research and Development, O=Tech Innovations Inc., L=San Francisco, ST=California, C=US"
+
+The VC contains all the attributes describing a given node, however, not all of them must be provided to the producer when purchasing a given resource.
+In fact, a given provider might require only a subset of them (e.g., only university and department are required).
+To this end, a **Verifiable Presentation (VP)** can be requested to the trusted issuer, including only the subset of requested attributes and preventing information disclosure (see Figure 3).
```

Comment:
There's no need to request a verifiable presentation to the issuer, each entity can generate it through **privacy and security manager** having a Verifiable Credential stored in his private wallet.